src/plugins/analysis/hash/code/hash.py

```diff
@@ -1,43 +1,134 @@
+from __future__ import annotations
+
+import contextlib
 import logging
+import sys
 from hashlib import algorithms_guaranteed
+from typing import TYPE_CHECKING, Optional
+
+import lief
+import ssdeep
+import tlsh
+from pydantic import BaseModel, Field
+from semver import Version
 
 import config
-from analysis.PluginBase import AnalysisBasePlugin
-from helperFunctions.hash import get_hash, get_imphash, get_ssdeep, get_tlsh
+from analysis.plugin import AnalysisPluginV0
+from analysis.plugin.compat import AnalysisBasePluginAdapterMixin
+from helperFunctions.hash import get_hash, get_md5
+
+if TYPE_CHECKING:
+    from io import FileIO
+
+ELF_MIME_TYPES = [
+    'application/x-executable',
+    'application/x-object',
+    'application/x-pie-executable',
+    'application/x-sharedlib',
+]
+
+
+class AnalysisPlugin(AnalysisPluginV0, AnalysisBasePluginAdapterMixin):
+    class Schema(BaseModel):
+        # The supported hashes are the ones from helperFunctions.hash and hashlib (except shake which does not make
+        # much sense with its variable length). If they are not supported on the platform or not selected in the
+        # configuration of the plugin, the value will be `None`. Only the md5 and sha256 hashes are guaranteed to be
+        # available (since they are needed down the line)
+
+        # from hashlib
+        md5: str = Field(description="md5 hash of the file's content")
+        sha256: str = Field(description="sha256 hash of the file's content")
+        sha1: Optional[str] = Field(description="sha1 hash of the file's content", default=None)
+        sha224: Optional[str] = Field(description="sha224 hash of the file's content", default=None)
+        sha384: Optional[str] = Field(description="sha384 hash of the file's content", default=None)
+        sha512: Optional[str] = Field(description="sha512 hash of the file's content", default=None)
+        blake2b: Optional[str] = Field(description="blake2b hash of the file's content", default=None)
+        blake2s: Optional[str] = Field(description="blake2s hash of the file's content", default=None)
+        sha3_224: Optional[str] = Field(description="sha3_224 hash of the file's content", default=None)
+        sha3_256: Optional[str] = Field(description="sha3_256 hash of the file's content", default=None)
+        sha3_384: Optional[str] = Field(description="sha3_384 hash of the file's content", default=None)
+        sha3_512: Optional[str] = Field(description="sha3_512 hash of the file's content", default=None)
+
+        # from helperFunctions.hash
+        ssdeep: Optional[str] = Field(description="ssdeep hash of the file's content", default=None)
+        tlsh: Optional[str] = Field(description="tlsh hash of the file's content", default=None)
+        imphash: Optional[str] = Field(
+            description="import hash of the executable's imported functions (only for ELF files)",
+            default=None,
+        )
```

Comment:
Two things:
1) I do not think this belongs in this plugin but rather in an elf specific plugin. But since this is status quo this is fine for now.
2) This needs some clarification. How exactly are the functions hashed? Which hash functions is used?

Reply:
I tried to improve the wording of the description to make it more specific